internal/contour/snapshot.go

```diff
@@ -0,0 +1,54 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package contour
+
+import (
+	xds "github.com/envoyproxy/go-control-plane/pkg/cache/types"
+	"github.com/envoyproxy/go-control-plane/pkg/cache/v2"
+	"github.com/projectcontour/contour/internal/dag"
+	"github.com/sirupsen/logrus"
+)
+
+// SnapshotHandler implements the xDS snapshot cache
+type SnapshotHandler struct {
```

Comment:
I'm not sure that "handler" is the right naming here. The "Handler" name is used for types that deal with Kubernetes events directly, but this is a DAG observer which is a different role.

Reply:
It "handles' snapshots? I dunno, please suggest I think the name is fine. =)

Reply:
> It "handles' snapshots? I dunno, please suggest I think the name is fine. =)

ROFL, it's just that it's a handler in a different sense than the other types that we name "*Handler".  I will get by with "handler" if we can't think of a evocative alternative :)